pybamm/solvers/ode_solver.py

```diff
@@ -63,7 +72,9 @@ def integrate(self, derivs, y0, t_eval, events=None, mass_matrix=None):
         events : method, optional
             A function that takes in t and y and returns conditions for the solver to
             stop
-        mass_matrix : :class:`pybamm.Matrix`
+        mass_matrix : array_like, optional
             The (sparse) mass matrix for the chosen spatial method.
+        jacobian : method, optional
+            A function that takes in t and y and returns the Jacobian
```

Comment:
are there any cases where `mass_matrix` depends on `t` and `y`?

Reply:
I can't think of any off the top of my head, but maybe there are some cases of this. Will probably be good to make `model.mass_matrix`a method sometime in the future as you suggest